chart/templates/connection-secret-job.yaml

```diff
@@ -0,0 +1,96 @@
+{{- $db := index .Values "timescaledb-single" -}}
+{{- if and $db.enabled (eq .Values.promscale.connectionSecretName "") (eq .Values.promscale.connection.password "") -}}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ .Release.Name }}-promscale-initializer-sa
+  namespace: {{ template "tobs.namespace" . }}
+  labels:
+    app: {{ template "tobs.fullname" . }}
+    chart: {{ template "tobs.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .Release.Name }}-promscale-initializer-role
+  namespace: {{ template "tobs.namespace" . }}
+  labels:
+    app: {{ template "tobs.fullname" . }}
+    chart: {{ template "tobs.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  verbs: ["get", "patch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: {{ .Release.Name }}-promscale-initializer-rolebinding
+  namespace: {{ template "tobs.namespace" . }}
+  labels:
+    app: {{ template "tobs.fullname" . }}
+    chart: {{ template "tobs.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: {{ .Release.Name }}-promscale-initializer-role
+subjects:
+- namespace: {{ template "tobs.namespace" . }}
+  kind: ServiceAccount
+  name: {{ .Release.Name }}-promscale-initializer-sa
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-connection-initializer
+  namespace: {{ template "tobs.namespace" . }}
+  labels:
+    app: {{ template "tobs.fullname" . }}
+    chart: {{ template "tobs.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+data:
+  password-initializer.sh: |
+    #!/bin/bash
+    while ! kubectl get secret {{ .Release.Name }}-credentials --namespace {{ template "tobs.namespace" . }}; do
+      echo "Waiting for {{ .Release.Name }}-credentials secret."
+      sleep 1
+    done
+    PASS=$(kubectl get secret --namespace {{ template "tobs.namespace" . }} {{ .Release.Name }}-credentials -o json | jq -r '.data["PATRONI_SUPERUSER_PASSWORD"]')
```

Comment:
can we avoid using an env variable here? Is that important for security?

Reply:
Which env variable, `PASS`, `PATRONI_SUPERUSER_PASSWORD`, or `PROMSCALE_DB_PASSWORD`? If you mean `PASS` then it is here to have a fairly readable script. If you mean `PATRONI_SUPERUSER_PASSWORD` then it is set in timescaledb helm chart and it is safely stored in kuberentes Secret. And lastly if you mean `PROMSCALE_DB_PASSWORD` then it is also stored as kubernetes Secret and it is part of promscale helm chart. 

Env variables set in Secrets are there for confidence and code clarity. Without that we would need to set them in Deployment which would be quite problematic.

Reply:
I mean PASS here. Is using a separate variable possibly creating a security hole? I am not sure

Reply:
AFAIK variable would need to be exported to be available outside the script, hence this shouldn't create a security hole.